Application/Frontend/pages/api/update-message-history.tsx

```diff
@@ -0,0 +1,60 @@
+import type { NextApiRequest, NextApiResponse } from 'next';
```

Comment:
- The function name handler is too generic and does not convey its purpose clearly. Consider using a more descriptive name that reflects its role, such as updateOrCreateMessageHistory.
- The error handling mechanism is well-implemented with appropriate error messages and status codes returned to the client. However, it would be beneficial to provide more specific error messages to aid in debugging, such as indicating which part of the operation failed.
- The code should include input validation to ensure that required fields (channelKey, messageHistory, owner, members) are present and have valid values. This helps prevent potential issues due to invalid or missing input.